sdk/src/azure/iot/az_iot_adu_client.c

```diff
@@ -0,0 +1,871 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <azure/iot/az_iot_adu_client.h>
+#include <azure/iot/az_iot_hub_client_properties.h>
+
+#include <azure/core/internal/az_log_internal.h>
+#include <azure/core/internal/az_precondition_internal.h>
+#include <azure/core/internal/az_result_internal.h>
+#include <stdio.h>
+
+/* Define the ADU agent component name.  */
+#define AZ_IOT_ADU_CLIENT_AGENT_COMPONENT_NAME "deviceUpdate"
+
+#define AZ_IOT_ADU_CLIENT_AGENT_INTERFACE_ID "dtmi:azure:iot:deviceUpdate;1"
+
+/* Define the ADU agent property name "agent" and sub property names.  */
```

Comment:
When do you decide some `#define` macros belong in the public header file and are publicly accessible, vs when they are "private implementation details" like this, and only exist in the source file?

We use `_az_` for such non-public `#defines`:
https://github.com/Azure/azure-sdk-for-c/blob/31e832e14a8e77f843639a02eca79de95309a454/sdk/src/azure/core/az_span.c#L22

https://github.com/Azure/azure-sdk-for-c/blob/5ad7985d50af8873c6148f867f57c487e294fb28/sdk/src/azure/core/az_http_policy.c#L47-L50

Reply:
I guess that's fair. Those could be `_az` since they aren't used by the developer. Only internal to the `.c` file

Reply:
Since it is just naming, as part of non-breaking implementation detail, it can be addressed later :)